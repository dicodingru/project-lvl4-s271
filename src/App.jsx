import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';

import { addMessage } from './actions';
import Root from './components/Root';

export default (store, socket) => {
  socket.on('newMessage', ({ data }) => {
    const { attributes } = data;
    store.dispatch(addMessage({ ...attributes }));
  });

  ReactDOM.render(
    <Provider store={store}>
      <Root />
    </Provider>,
    document.getElementById('chat')
  );
};
