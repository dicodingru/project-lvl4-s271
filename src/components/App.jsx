import React from 'react';
import PropTypes from 'prop-types';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { addMessage } from '../actions';
import rootReducer from '../reducers';
import Root from './Root';

/* eslint-disable no-underscore-dangle */
const reduxDevtools = window.__REDUX_DEVTOOLS_EXTENSION__;
/* eslint-enable */

const App = ({ initialState, socket }) => {
  const store = createStore(
    rootReducer,
    initialState,
    reduxDevtools && reduxDevtools()
  );

  socket.on('newMessage', ({ data }) => {
    const { attributes } = data;
    store.dispatch(addMessage({ ...attributes }));
  });

  return (
    <Provider store={store}>
      <Root />
    </Provider>
  );
};

App.propTypes = {
  initialState: PropTypes.shape({
    channels: PropTypes.array,
    messages: PropTypes.array
  }).isRequired,
  socket: PropTypes.instanceOf(WebSocket).isRequired
};

export default App;
