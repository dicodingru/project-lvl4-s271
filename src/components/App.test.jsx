import React from 'react';
import App from './App';

test('App', () => {
  const wrapper = mount(
    <App
      initialState={{
        channels: [
          { id: 1, name: 'general', removable: false },
          { id: 2, name: 'random', removable: false }
        ]
      }}
    />
  );

  expect(wrapper.render()).toMatchSnapshot();
});
