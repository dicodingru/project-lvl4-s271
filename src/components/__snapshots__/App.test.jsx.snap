// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
<ul>
  <li>
    name
  </li>
  <li>
    name
  </li>
</ul>
`;
